docker_ci_deploy/tests/test_main.py

```diff
@@ -380,6 +426,181 @@ def test_version_new_tag_with_version(self, capfd):
             'push test-image:1.2.3-def',
         ])
 
+    # FIXME?: The following 2 tests describe a weird, unintuitive edge case :-(
+    # Passing `--tag latest` with `--tag-version <version>` but *not*
+    # `--tag-latest` doesn't actually get you the tag 'latest' but rather
+    # effectively removes any existing tag.
```

Comment:
Yeah I did consider special-casing it.. the logic just gets pretty messy then (and it's already not fantastic)